### ISSUE TITLE
Round ended event for bot is not sent

### ISSUE THREAD

**Issue by SirStone** [robocode-dev/tank-royale#60]
**Describe the bug**
The new [round ended event for bot](https://github.com/robocode-dev/tank-royale/blob/master/schema/schemas/round-ended-event-for-bot.yaml) is never received by my bot. I believe that even the no-more "round ended event" was never sent to bots.

**Expected behavior**
at the end of each round a [round ended event for bot](https://github.com/robocode-dev/tank-royale/blob/master/schema/schemas/round-ended-event-for-bot.yaml) is received by the bot

**Comment by flemming-n-larsen** [comment]
I will have a look into this. 👍

**Comment by flemming-n-larsen** [comment]
Observation: With one battle, the RoundEndedEvent never occurs. With 10 rounds, it does occur for all rounds, except the last one, it seems.

**Comment by flemming-n-larsen** [comment]
@SirStone 

I did find a bug with the RoundEventEvent not being sent by the server when the game has ended. It should send this, even when the game has just ended, of course.

But when I examine the [round-ended-event-for-bot](https://github.com/robocode-dev/tank-royale/blob/master/schema/schemas/round-ended-event-for-bot.yaml) on the Java and C# bot APIs, my conclusion is that the onRoundEnded(RoundEndedEvent) is definitely being called, and is working fine with the existing bot APIs. You can easily see this when you add this print line to e.g. the Corners robot in the Java or C# sample bots:

Java:
```java
    @Override
    public void onRoundEnded(RoundEndedEvent e) {
        System.out.println("onRoundEnded: " + e.getTurnNumber());
    }
```

C#:
```csharp
    public override void OnRoundEnded(RoundEndedEvent e) {
        Console.WriteLiine("OnRoundEnded: " + e.TurnNumber);
    }
```

And the WebSocket text handler also receives and handles the message:

Line 710 in [BaseBotInternals.java](https://github.com/robocode-dev/tank-royale/blob/master/bot-api/java/src/main/java/dev/robocode/tankroyale/botapi/internal/BaseBotInternals.java)

```java
                    switch (dev.robocode.tankroyale.schema.Message.Type.fromValue(type)) {
                        ...
                        case ROUND_ENDED_EVENT_FOR_BOT:
                            handleRoundEnded(jsonMsg);
                            break;
                            ...
```

Are you sure the  [round-ended-event-for-bot](https://github.com/robocode-dev/tank-royale/blob/master/schema/schemas/round-ended-event-for-bot.yaml) is not received by the WebSocket client with your API?

**Comment by SirStone** [comment]
I've checked deeper my code and I found out that I am in error, the "RoundEndedEventForBot" is sent indeed!
My mistake was in my NIM code about the ENUM type used toghether with a SWITCH CASE that I failed to understand how it works 100%. I've now fixed the code for this message and I will not fall in the same mistake in the future.

At least looks like you found a bug on your own while  checking the validity of mine, but still, sorry for the false alert.
